### PR TITLE
Issue/1787

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #1787 

## How Has This Been Tested?
- Added **PayPal IPN Notifications** status under System Info.
- Tested using multiple PayPal Standard transactions with `complete` and `pending` statuses.
    - When status is pending, then donation note will be `N/A`
    - When status is complete, then donation note will be `N/A`  > `Last IPN Received` > `PayPal Transaction ID` > `Status change from Pending to Complete`

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/tools/views/html-admin-page-system-info.php [hunk 1]
```diff
@@ -489,6 +489,11 @@
 			<td><?php echo esc_html( ! empty( $default_gateway ) ? $default_gateway : '&ndash;' ); ?></td>
 		</tr>
 		<tr>
+			<td data-export-label="PayPal IPN Notifications"><?php _e( 'PayPal IPN Notifications', 'give' ); ?>:</td>
+			<td class="help"><span class="give-tooltip give-icon give-icon-question" data-tooltip="<?php echo esc_attr( __( 'Whether admin has enabled PayPal IPN notifications or not.', 'give' ) ); ?>"></span></td>
+			<td><?php echo 'enabled' === give_get_option( 'paypal_verification' ) ? __( 'Enabled', 'give' ) : __( 'Disabled', 'give' ); ?></td>
```
Comment (DevinWalker) [inline includes/admin/tools/views/html-admin-page-system-info.php]: This isn't correct because this is simply checking whether the `paypal_verification` if active, _not_ whether an IPN has actually been received or not. 

Please re-read over my scenarios comment on #1787 where I describe how this should work. If you have questions, we can hop on a call. Basically, you need to flag an option once an IPN _has been_ received.

File: includes/gateways/paypal-standard.php [hunk 2]
```diff
@@ -51,6 +51,9 @@ function give_process_paypal_payment( $payment_data ) {
 	give_validate_nonce( $payment_data['gateway_nonce'], 'give-gateway' );
 	$payment_id = give_create_payment( $payment_data );
 
+	// Register message in donation note that PayPal IPN is not received yet.
+	give_insert_payment_note( $payment_id, __( 'N/A', 'give' ) );
```
Comment (DevinWalker) [inline includes/gateways/paypal-standard.php]: This info wouldn't be particularly useful, remove.

File: includes/gateways/paypal-standard.php [hunk 3]
```diff
@@ -253,6 +256,9 @@ function give_process_paypal_web_accept( $data, $payment_id ) {
 		return;
 	}
 
+	// Register message in donation note that IPN received successfully.
+	give_insert_payment_note( $payment_id, __( 'Last IPN Received', 'give' ) );
```
Comment (DevinWalker) [inline includes/gateways/paypal-standard.php]: This is not particularly useful either. To be useful it should just "IPN received for {transaction ID}. Status {auth_status}" - See https://developer.paypal.com/docs/classic/ipn/integration-guide/IPNandPDTVariables/#transaction-and-notification-related-variables

Have you used Postman before to spoof the IPN? I can help you with this if not.

File: give.php [hunk 4]
```diff
@@ -5,7 +5,7 @@
  * Description: The most robust, flexible, and intuitive way to accept donations on WordPress.
  * Author: WordImpress
  * Author URI: https://wordimpress.com
- * Version: 1.8.11
+ * Version: 1.8.12
```
Comment (ravinderk) [inline give.php]: @mehul0810 version number update should be in different PR

File: includes/admin/tools/views/html-admin-page-system-info.php [hunk 5]
```diff
@@ -489,6 +494,29 @@
...
+			<td data-export-label="PayPal IPN Notifications"><?php _e( 'PayPal IPN Notifications', 'give' ); ?>:</td>
+			<td class="help"><span class="give-tooltip give-icon give-icon-question" data-tooltip="<?php echo esc_attr( __( 'Displays whether when last PayPal IPN is received with which donation or transaction.', 'give' ) ); ?>"></span></td>
+			<td>
+				<?php
+				$last_paypal_ipn_received = get_option( 'give_last_paypal_ipn_received' );
+				if( is_array( $last_paypal_ipn_received ) && count( $last_paypal_ipn_received ) > 0 ) {
+					$transaction_information_url = 'https://history.paypal.com/cgi-bin/webscr?cmd=_history-details-from-hub&id=' . $last_paypal_ipn_received['transaction_id'];
+					$donation_url = site_url() . '/wp-admin/edit.php?post_type=give_forms&page=give-payment-history&view=view-payment-details&id=' . $last_paypal_ipn_received['payment_id'];
+					echo sprintf(
+							__( 'IPN received for <a href="%s">#%s</a> ( <a href="%s" target="_blank">%s</a> ). Status %s', 'give' ),
```
Comment (ravinderk) [inline includes/admin/tools/views/html-admin-page-system-info.php]: @mehul0810 Can we show time also here?

Comment (mehul0810) [inline includes/admin/tools/views/html-admin-page-system-info.php]: @ravinderk I've added date and time in System Info same as donation notes

File: includes/admin/tools/views/html-admin-page-system-info.php [hunk 6]
```diff
@@ -451,6 +451,11 @@
 			<td><?php echo ! empty( $give_options['failure_page'] ) ? esc_url( get_permalink( $give_options['failure_page'] ) ) : '&ndash;'; ?></td>
 		</tr>
 		<tr>
+			<td data-export-label="Donation History Page"><?php _e( 'Donation History Page', 'give' ); ?>:</td>
+			<td class="help"><span class="give-tooltip give-icon give-icon-question" data-tooltip="<?php echo esc_attr( __( 'The page where past donations are listed.', 'give' ) ); ?>"></span></td>
```
Comment (ravinderk) [inline includes/admin/tools/views/html-admin-page-system-info.php]: @DevinWalker Can you check tooltip description?

Comment (DevinWalker) [inline includes/admin/tools/views/html-admin-page-system-info.php]: Looks good to me

**Comment by mehul0810** [comment]
For Reference, This is how it looks after the changes:

### Donation Notes
image

### System Info
image

**Comment by DevinWalker** [comment]
@mehul0810 please add an addition option for the PayPal IPN Verification option as we discussed on the call this evening. :)

**Comment by mehul0810** [comment]
@DevinWalker I've added an additional option for the status of PayPal IPN verification status under `System Info` as per the discussion over call today.

**Comment by DevinWalker** [comment]
Thanks @mehul0810
